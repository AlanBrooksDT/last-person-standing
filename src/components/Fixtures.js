import { useState, useEffect } from "react";
import axios from "axios";
import '../styles/Fixtures.css';
import dayjs from 'dayjs'
import advancedFormat from "dayjs/plugin/advancedFormat"

dayjs.extend(advancedFormat)

const Fixtures = () => {
  const [fixtures, setFixtures] = useState([]);
  const [week, setWeek] = useState(0);

  console.log("Fixtures");
    
  useEffect(() => {
    const getWeek = () => {
     axios.get(`http://api.football-data.org/v2/competitions/2021`)
        .then((response) => {
          setWeek(response.data.currentSeason.currentMatchday)
        })
        .catch((err) => {});
    };
    getWeek();
  }, []);

  console.log(week);
  
  useEffect(() => {
  
    const getFixtures = () => {
      axios.get(`http://api.football-data.org/v2/competitions/2021/matches?matchday=${week}`)
        .then((response) => {
          setFixtures(response.data.matches)
        })
        .catch((err) => {});
    };
    getFixtures();
  }, [week]);

  axios.interceptors.request.use(
    config => {
      config.headers["X-Auth-Token"] = `1c195cb9649b46678c999c7bf128cddd`;
      return config;
    },
    err => {
    
    }
  );
  
  console.log(fixtures)

  const renderMatchResults = () => {

    return (

        <>
          {fixtures.map((match, i) => (
            <div key={i} className="matchContainer">
              <div className="match">
                <div className="homeTeam">{match.homeTeam.name}</div>
                <div className="score">
                  {match.score.fullTime.homeTeam} -{" "}
                  {match.score.fullTime.awayTeam}
                </div>
                <div className="awayTeam">{match.awayTeam.name}</div>
              </div>
            </div>
          ))}
        </>
      )
  }
    return (
      <div className="fixtureContainer">
        Current Week: {week}
        {renderMatchResults()}

      </div>
    );
  }
  

export default Fixtures;
