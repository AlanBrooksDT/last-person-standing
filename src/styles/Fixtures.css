* {
  box-sizing: border-box;
}

.fixtureContainer {
  margin: 0;
  margin-top: 40px;
  padding: 40px;
  width: calc(-10vw + 70%);
  display: block;
  overflow: auto;
  margin-left: auto;
  margin-right: auto;
  background-color: rgba(0, 0, 0, 0.8);
  color: white;
}

.matchContainer {
  border: 1px solid #ccc;
  display: table;
  table-layout: fixed;
  width: 100%;
  will-change: contents;
  text-align: center;
}

.matchContainer.POSTPONED {
  background-color: #d6181f;
}

.match {
  display: table;
  table-layout: fixed;
  width: 100%;
  height: 100%;
  padding: 10px;
}

.teamSelection {
  border: 1px solid #ccc;
  display: table;
  width: 100%;
  height: 100%;
  padding: 3px;
  margin-top: 10px;
}

.selectionHeader {
  display: table-cell;
  text-align: left;
  vertical-align: middle;
  text-overflow: ellipsis;
  width: 20%;
  padding-left: 10px;
}

.teamSelected {
  display: table-cell;
  text-align: center;
  vertical-align: middle;
  text-overflow: ellipsis;
  width: 80%;
  font-size: 2rem;
}

.matchDate {
  display: table;
  height: 100%;
}

.score {
  display: table-cell;
  vertical-align: middle;
  width: 60px;
  table-layout: auto;
  background: #d6181f;
  color: #fff;
  margin: auto;
  background-clip: content-box;
  border-left: 10px solid transparent;
  border-right: 10px solid transparent;
  border-top: 5px solid transparent;
  border-bottom: 5px solid transparent;
}

.score.FINISHED {
  background-color: green;
}

.homeTeam {
  display: table-cell;
  text-align: right;
  vertical-align: middle;
  text-overflow: ellipsis;
  width: 35%;
}

.homeTeam:hover {
  cursor: pointer;
}

.awayTeam {
  display: table-cell;
  vertical-align: middle;
  text-align: left;
  text-overflow: ellipsis;
  width: 35%;
}

.awayTeam:hover {
  cursor: pointer;
}

.matchStatus {
  display: table-cell;
  text-align: right;
  vertical-align: middle;
  width: 60px;
}

@media (max-width: 868px) {
  .fixtureContainer {
    width: 95%;
  }
}

@media (max-width: 540px) {
  .fixtureContainer {
    width: 100%;
    padding: 10px;
  }

  .score {
    width: 50px;
    border-top: 10px solid transparent;
    border-bottom: 10px solid transparent;
  }
}
