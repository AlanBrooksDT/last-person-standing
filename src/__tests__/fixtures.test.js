import React from "react";
import { render } from "@testing-library/react";
import Fixtures from "../components/Fixtures";

describe("Fixtures", () => {
  it("renders", () => {
    const { asFragment, getByTestId } = render(<Fixtures />);

    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("fixtureContainer")).toBeInTheDocument();
    expect(getByTestId("homeTeam")).toBeInTheDocument();
  });
});
