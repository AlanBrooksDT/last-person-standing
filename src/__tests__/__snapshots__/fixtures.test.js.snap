// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixtures renders 1`] = `
<DocumentFragment>
  <div
    class="fixtureContainer"
    data-testid="fixtureContainer"
  />
</DocumentFragment>
`;
